File: java/src/prep/leetcode/RightSideView.java
package prep.leetcode;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class RightSideView 
{
	class TreeNode 
	{
		 int val;
		 TreeNode left;
		 TreeNode right;
		 TreeNode() {}
		 TreeNode(int val) 
		 { 
			 this.val = val; 
		 }
		 
		 TreeNode(int val, TreeNode left, TreeNode right) 
		 {
			 this.val = val;
			 this.left = left;
			 this.right = right;
		 }
	 }
	
	public List<Integer> rightSideView(TreeNode root) {
        List<Integer> right_view = new ArrayList<Integer>();
        Queue<TreeNode> q = new LinkedList<TreeNode>();

        if(root==null) return right_view;
        q.offer(root);
        while(!q.isEmpty())
        {
            int size = q.size();
            for(int i=0;i<size;i++)
            {
                TreeNode curr = q.poll();
                if(i==0) right_view.add(curr.val);
                if(curr.right!=null)q.offer(curr.right);
                if(curr.left!=null)q.offer(curr.left);
            }
            
        }
        return right_view;
    }

}

File: python/tree/RightSideView.py
from typing import List
from typing import Optional
from collections import deque
from tree import TreeNode


class RightSideView:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        if not root:
            return res

        q = deque()
        q.append(root)

        while (len(q) > 0):
            size = len(q)
            for i in range(size):
                curr = q.popleft()
                if i == 0:
                    res.append(curr.val)

                if curr.right:
                    q.append(curr.right)
                if curr.left:
                    q.append(curr.left)

        return res